lib/BuildSystem/BuildSystemExtensionManager.cpp

```diff
@@ -92,6 +94,7 @@ BuildSystemExtensionManager::lookupByCommandPath(StringRef path) {
     return {};
   }
 
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
```

Comment:
Wouldn't `#if _WIN32 #else` be simpler? It's probably the only OS of any relevance that *doesn't* have the POSIX dl* APIs.

Also, can you please put the LoadLibrary/dlopen, FreeLibrary/dlclose, GetProcAddress/dlsym into PlatformSupport? They're 1:1 analogues and it would reduce the #ifdef-ery at the call sites.

Also, remember, no *A APIs, only *W :)

Reply:
I was attempting to avoid the wcstombcs round tripping (particularly since the file path up until this point has been passed as `char *`) ... but, sure, can do the conversion.

Reply:
You fixed the *A vs *W API usage but can you please address the first part of my comment as well?